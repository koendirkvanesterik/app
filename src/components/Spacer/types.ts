export type Props = {
  /**
   * Union type determining margin size
   */
  size?: 'lg' | 'md' | 'sm'
}
