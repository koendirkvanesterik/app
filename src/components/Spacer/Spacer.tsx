import classNames from 'classnames'
import React from 'react'
import { Props } from './types'
import { getMarginBy } from './utils'

export const Spacer: React.FunctionComponent<Props> = ({ size = 'sm' }) => (
  <div className={classNames([getMarginBy(size)])} />
)
