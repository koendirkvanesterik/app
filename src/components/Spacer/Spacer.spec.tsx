import { render } from '@testing-library/react'
import React from 'react'
import { Spacer } from './Spacer'

it('should render as expected', () => {
  const { container } = render(<Spacer />)
  expect(container).toMatchSnapshot()
})
