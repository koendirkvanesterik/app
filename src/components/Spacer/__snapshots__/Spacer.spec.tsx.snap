// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div>
  <div
    class="mt-40"
  />
</div>
`;
